### PR TITLE
Integrate Phala World Pallet

### PR DESCRIPTION
**Pull request by HashWarlock**
## Targets
- [x] Integrate Uniques Pallet
- [x] Test Functions in Uniques to Ensure BaseFilter Works
- [x] Integrate RMRK Pallet
- [x] Test Functions in RMRK to Ensure BaseFilter Works 
- [x] Integrate Phala World Pallet
- [x] Add Metadata URI from Decentralized Storage for Origin of Shell NFTs
- [x] Implement Privileged Create Collection in Phala World Pallet
- [x] Implement a Giveaway Function for the Giveaway NFTs
- [x] Run Tests on Thala
- [ ] Prepare on Rhala for Staging
- [ ] Prepare for Deployment on Khala

## Procedure
- Integrate Uniques pallet into thala, rhala and khala runtimes
  - Whitelist functions needed in Uniques using BaseFilter
- Integrate RMRK pallet into thala, rhala and khala runtimes
  - Whitelist functions needed in RMRK using BaseFilter
- Integrate Phala World Pallet into thala, rhala and khala runtimes
  - Run tests to ensure Phala World deployment works

### CODE REVIEW

File: pallets/phala-world/src/nft_sale.rs [hunk 1]
```diff
@@ -0,0 +1,1614 @@
...
+		/// Parameters:
+		/// - origin: Expected to be called by `OverlordOrigin`
+		/// - new_overlord: T::AccountId
+		#[pallet::weight(0)]
+		pub fn set_overlord(
+			origin: OriginFor<T>,
+			new_overlord: T::AccountId,
+		) -> DispatchResultWithPostInfo {
+			// This is a public call, so we ensure that the origin is some signed account.
+			T::OverlordOrigin::ensure_origin(origin)?;
```
Comment (jasl) [inline pallets/phala-world/src/nft_sale.rs:772]: What if there hasn't a overloard yet?

File: runtime/rhala/src/lib.rs [hunk 2]
```diff
@@ -780,6 +825,106 @@ impl cumulus_pallet_parachain_system::Config for Runtime {
...
+    pub const MagicOriginOfShellPrice: Balance = 10_000 * DOLLARS;
+    pub const PrimeOriginOfShellPrice: Balance = 500 * DOLLARS;
+    pub const IterLimit: u32 = 1_000;
+    pub const FoodPerEra: u32 = 5;
+    pub const MaxFoodFeedSelf: u8 = 2;
+    pub const IncubationDurationSec: u64 = 1_209_600;
+}
+impl pallet_pw_nft_sale::Config for Runtime {
+    type Event = Event;
+    type OverlordOrigin = EnsureRootOrHalfCouncil;
```
Comment (jasl) [inline runtime/rhala/src/lib.rs:910]: Not blocking
Maybe we should call another name, e.g. `RootOrigin` or `GovernanceOrigin`
I was confused in `set_overlord`, why set overlord requires `OverlordOrigin`?

Comment (HashWarlock) [reply runtime/rhala/src/lib.rs:910]: I can change this to `GovernanceOrigin`

**Comment by jasl** [comment]
Will deployed to PC-3 testnet (thala) tomorrow

**Pull request merged**